ci-operator/step-registry/upi/libvirt/install/upi-libvirt-install-commands.sh

```diff
@@ -0,0 +1,244 @@
+MAC_ADDRESS=$(leaseLookup "bootstrap[0].mac")
+clone_volume ${NODE}-volume
+create_vm ${NODE} ${MAC_ADDRESS} bootstrap
+
+# Create the control plane nodes.
+for (( i=0; i<=${CONTROL_COUNT}-1; i++ )); do
+  NODE="${LEASED_RESOURCE}-control-${i}"
+  MAC_ADDRESS=$(leaseLookup "control-plane[$i].mac")
+  clone_volume ${NODE}-volume
+  create_vm ${NODE} ${MAC_ADDRESS} master
+done
+
+# Create the compute nodes.
+for (( i=0; i<=${COMPUTE_COUNT}-1; i++ )); do
+  NODE="${LEASED_RESOURCE}-compute-${i}"
+  MAC_ADDRESS=$(leaseLookup "compute[$i].mac")
+  clone_volume ${NODE}-volume
+  create_vm ${NODE} ${MAC_ADDRESS} worker
+done
+
+date "+%F %X" > "${SHARED_DIR}/CLUSTER_INSTALL_START_TIME"
+
+${OCPINSTALL} --dir "/tmp" wait-for bootstrap-complete &
+# TODO: collect logs in case of failure
+wait "$!"
+
+# Sleep between destroy and undefine to allow for a slight destroy lag
+${VIRSH} destroy "${LEASED_RESOURCE}-bootstrap"
+sleep 1s
+${VIRSH} undefine "${LEASED_RESOURCE}-bootstrap"
+
+echo "Approving pending CSRs"
+approve_csrs () {
+  oc version --client
+  while true; do
+    if [[ ! -f /tmp/install-complete ]]; then
+      # even if oc get csr fails continue
+      oc get csr -ojson | yq-v4 -oy '.items[] | select(.status | length == 0) | .metadata.name' | xargs --no-run-if-empty oc adm certificate approve || true
+      sleep 15 & wait
+      continue
+    else
+      break
+    fi
+  done
+}
+approve_csrs &
+
+# Add a small buffer before waiting for install completion
+sleep 5m
+
+set +x
+echo "Completing UPI setup"
+${OCPINSTALL} --dir="/tmp" wait-for install-complete 2>&1 | grep --line-buffered -v password &
+wait "$!"
+
+# Check for image registry availability
+for i in {1..10}; do
+  count=$(oc get configs.imageregistry.operator.openshift.io/cluster --no-headers | wc -l)
+  echo "Image registry count: ${count}"
+  if [[ ${count} -gt 0 ]]; then
+    break
+  fi
+  sleep 30
+done
+
+# Patch the image registry
+echo "Patching image registry..."
+oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"storage":{"emptyDir":{}}, "managementState": "Managed"}}'
+sleep 60
+for i in {1..30}; do
+  READY_REPLICAS=$(oc get deployment -n openshift-image-registry -oyaml image-registry | yq-v4 -oy ".status.readyReplicas")
+  TOTAL_REPLICAS=$(oc get deployment -n openshift-image-registry -oyaml image-registry | yq-v4 -oy ".status.replicas")
+  if [[ "${READY_REPLICAS}" == "${TOTAL_REPLICAS}" ]]; then
+    echo "Patched successfully!"
+    break
+  fi
+  sleep 15
+done
+
+date "+%F %X" > "${SHARED_DIR}/CLUSTER_INSTALL_END_TIME"
+
+# Password for the cluster gets leaked in the installer logs and hence removing them.
+sed -i 's/password: .*/password: REDACTED"/g' /tmp/.openshift_install.log
+cp /tmp/.openshift_install.log "${SHARED_DIR}"/.openshift_install.log
+
+# Save the kubeconfig again to make sure any changes during install are captured in future steps
+save_credentials
+
+touch /tmp/install-complete
```

Comment:
What uses this? Historically we would do stuff like this in templates because other containers were busy-waiting on the files appearing, but I don't think this is needed anymore.